python/tank/path_cache.py

```diff
@@ -572,250 +565,417 @@ def _do_full_sync(self, cursor):
+            imported_data = self._import_filesystem_location_entry(cursor, x)
+            if imported_data:
+                return_data.append(imported_data)
 
-            # if the path cannot be split up into a root_name and a leaf path
-            # using the roots.yml file, log a warning and continue. This can happen
-            # if roots files and storage setups change half-way through a project,
-            # or if roots files are not in sync with the main storage definition
-            # in this case, we want to just warn and skip rather than raise
-            # an exception which will stop execution entirely.
-            try:
-                root_name, relative_path = self._separate_root(local_os_path)
-            except TankError, e:
-                log.debug("Could not resolve storages - skipping: %s" % e)
-                continue
-            
-            # all validation checks seem ok - go ahead and make the changes.
-            new_rowid = self._add_db_mapping(cursor, local_os_path, entity, is_primary)
-            if new_rowid:
-                # something was inserted into the db!
-                # because this record came from shotgun, insert a record in the
-                # shotgun_status table to indicate that this record exists in sg
-                cursor.execute("INSERT INTO shotgun_status(path_cache_id, shotgun_id) "
-                               "VALUES(?, ?)", (new_rowid, x["id"]) )
-            
-                # and add this entry to our list of new things that we will return later on.
-                return_data.append({"entity": entity, 
-                                    "path": local_os_path, 
-                                    "metadata": SG_METADATA_FIELD})
-            
-            else:
-                # Note: edge case - for some reason there was already an entry in the path cache
-                # representing this. This could be because of duplicate entries and is
-                # not necessarily an anomaly. It could also happen because a previos sync failed
-                # at some point half way through.
-                log.debug("Found existing record for '%s', %s. Skipping." % (local_os_path, entity))
-            
-        # lastly, id of this event log entry for purpose of future syncing
+        # lastly, save the id of this event log entry for purpose of future syncing
         # note - we don't maintain a list of event log entries but just a single
         # value in the db, so start by clearing the table.
-        log.debug("Inserting path cache marker %s in the sqlite db" % max_event_log_id)
-        cursor.execute("DELETE FROM event_log_sync")
-        cursor.execute("INSERT INTO event_log_sync(last_id) VALUES(?)", (max_event_log_id, ))
-            
+
+        self._update_last_event_log_synced(cursor, max_event_log_id)
+
         self._connection.commit()
 
         return return_data
 
+    def _update_last_event_log_synced(self, cursor, event_log_id):
```

Comment:
This is pretty much code that was originally in `_replay_folder_entities`, but it has been refactored into its own thing so it can be reused during incremental sync as well.